From 04e1331b93df0c1a4c59f9067fd02eb384a8b9e6 Mon Sep 17 00:00:00 2001
From: Ovidiu Panait <ovidiu.panait.rb@renesas.com>
Date: Wed, 8 Apr 2026 09:41:49 +0000
Subject: [PATCH 06379/12543] arm64: dts: renesas: r9a09g056: Use syscon
 compatible for the system controller

The Renesas RZ/V2N TSU driver needs the system controller syscon
to be able to retrieve calibration data. For this, it uses
syscon_regmap_lookup_by_phandle_args(). But, in v6.12 CIP,
syscon_regmap_lookup_by_phandle_args() returns ERR_PTR(-EINVAL) if
the node for which it is called doesn't have the syscon compatible.

Therefore, add syscon compatible for the system controller.

Signed-off-by: Ovidiu Panait <ovidiu.panait.rb@renesas.com>
Reviewed-by: Pavel Machek <pavel@nabladev.com>
Signed-off-by: Nobuhiro Iwamatsu <nobuhiro.iwamatsu.x90@mail.toshiba>
---
 arch/arm64/boot/dts/renesas/r9a09g056.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/renesas/r9a09g056.dtsi b/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
index a08d10950651..0d60d4cdb257 100644
--- a/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
+++ b/arch/arm64/boot/dts/renesas/r9a09g056.dtsi
@@ -292,7 +292,7 @@ cpg: clock-controller@10420000 {
 		};
 
 		sys: system-controller@10430000 {
-			compatible = "renesas,r9a09g056-sys";
+			compatible = "renesas,r9a09g056-sys", "syscon";
 			reg = <0 0x10430000 0 0x10000>;
 			clocks = <&cpg CPG_CORE R9A09G056_SYS_0_PCLK>;
 			resets = <&cpg 0x30>;
-- 
2.53.0

